### PR TITLE
Feat/fip 24 gas rebates

### PR DESCRIPTION
**Pull request by ditchfieldcaleb**

### CODE REVIEW

File: proposals/dao/fip_24.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+// Note that these mock the on-chain proposal, but the actual execution will be handled by the governor
+async function run(addresses, oldContracts, contracts, logging) {
+    const {
+        fei,
+        tribalChiefOptimisticTimelock,
+        ethLidoPCVDeposit,
+        oldEthReserveStabilizer
+    } = contracts;
+
+    await fei.mint(tribalChiefOptimisticTimelock.address, '100000000000000000000');
```
Comment (Joeysantoro) [inline proposals/dao/fip_24.js:34]: nit: elsewhere we define an `e18` variable with 18 zeros and use string interpolation to make these more readable

File: proposals/dao/fip_24.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    await oldEthReserveStabilizer.withdraw("0x35ED000468f397AA943009bD60cc6d2d9a7d32fF", '5000000000000000000');
+    await oldEthReserveStabilizer.withdraw(ethLidoPCVDeposit.address, '3979000000000000000000');
```
Comment (Joeysantoro) [inline proposals/dao/fip_24.js:36]: Can we be exact here and assert 0 ETH remain?

File: proposals/dao/fip_24.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        ethLidoPCVDeposit,
+        oldEthReserveStabilizer
+    } = contracts;
+
+    // Optimistic timelock should have 1 million FEI
+    const optimisticTimeLockFEIBalance = await fei.balanceOf(tribalChiefOptimisticTimelock.address);
+    assert.equal(optimisticTimeLockFEIBalance.toString(), '100000000000000000000');
+
+    // Optimistic multisig should have 50 eth
+    const optimisticMultisigETHBalance = await web3.eth.getBalance("0x35ED000468f397AA943009bD60cc6d2d9a7d32fF");
```
Comment (Joeysantoro) [inline proposals/dao/fip_24.js:58]: What contract is "0x35ED000468f397AA943009bD60cc6d2d9a7d32fF"? Should be read in from addresses param

File: proposals/dao/fip_24.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    assert.equal(optimisticMultisigETHBalance.toString(), '5000000000000000000');
+
+    // EthLidoPCVDeposit should have NO eth
+    const ethLidoPCVDepositETHBalance = await web3.eth.getBalance(ethLidoPCVDeposit.address);
+    assert.equal(ethLidoPCVDepositETHBalance.toString(), '0');
+
+    // Old eth reserve stablizier should have ~no eth
+    const oldETHReserveStabilizerETHBalance = await web3.eth.getBalance(oldEthReserveStabilizer.address);
```
Comment (Joeysantoro) [inline proposals/dao/fip_24.js:66]: Should be 0

File: proposals/description/fip_24.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+    "proposal_title": "FIP-24 Gas Refund for TribalChief Transition",
```
Comment (Joeysantoro) [inline proposals/description/fip_24.json:2]: Can you add a proposal text file too?

File: proposals/description/fip_24.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                "1000000000000000000000000"
+            ],
+            "description": "Mint 1 million FEI to the optimistic timelock"
+        },
+        {
+            "address": "0xa08A721dFB595753FFf335636674D76C455B275C",
+            "values": "0",
+            "method": "withdraw(address,uint256)",
+            "arguments": [
+                "0x35ED000468f397AA943009bD60cc6d2d9a7d32fF",
```
Comment (Joeysantoro) [inline proposals/description/fip_24.json:20]: Should we send the ETH to timelock too?

**Comment by Joeysantoro** [comment]
Overall looks good left a few comments

File: proposals/dao/fip_24.js (hunk 1 above)
Comment (ditchfieldcaleb) [reply proposals/dao/fip_24.js:34]: Fixed

File: proposals/dao/fip_24.js (hunk 2 above)
Comment (ditchfieldcaleb) [reply proposals/dao/fip_24.js:36]: Done.

File: proposals/dao/fip_24.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+async function run(addresses, oldContracts, contracts, logging) {
+    const {
+        fei,
+        tribalChiefOptimisticTimelock,
+        ethLidoPCVDeposit,
+        oldEthReserveStabilizer
+    } = contracts;
+
+    await fei.mint(tribalChiefOptimisticTimelock.address, '100000000000000000000');
+    await oldEthReserveStabilizer.withdraw("0x35ED000468f397AA943009bD60cc6d2d9a7d32fF", '5000000000000000000');
```
[the remark this answers is not in the document]
Comment (ditchfieldcaleb) [reply proposals/dao/fip_24.js:35]: Fixed.

File: proposals/dao/fip_24.js (hunk 3 above)
Comment (ditchfieldcaleb) [reply proposals/dao/fip_24.js:58]: Fixed.

File: proposals/dao/fip_24.js (hunk 4 above)
Comment (ditchfieldcaleb) [reply proposals/dao/fip_24.js:66]: Fixed

File: proposals/description/fip_24.json (hunk 6 above)
Comment (ditchfieldcaleb) [reply proposals/description/fip_24.json:20]: I honestly don't see a reason to

File: proposals/description/fip_24.json (hunk 5 above)
Comment (ditchfieldcaleb) [reply proposals/description/fip_24.json:2]: Done.

File: proposals/description/fip_24.txt [hunk 8]
```diff
@@ -0,0 +1,21 @@
+FIP-24 Gas Refund for TribalChief Transition
+
+This proposal will provide gas refunds in the form of FEI to eligible addresses.
```
Comment (Joeysantoro) [inline proposals/description/fip_24.txt:3]: Add a note saying this just funds the multisig to carry out the refunds

**Review by Joeysantoro — APPROVED** [review]
LGTM

File: proposals/description/fip_24.txt (hunk 8 above)
Comment (ditchfieldcaleb) [reply proposals/description/fip_24.txt:3]: Done.

**Pull request merged**